experimental/listener.go

```diff
@@ -6,6 +6,24 @@ import (
 	"github.com/tetratelabs/wazero/api"
 )
 
+// StackIterator allows iterating on each function of the call stack, starting
+// from the top. At least one call to Next() is required to start the iteration.
+//
+// Example:
+//
+//	for it.Next() {
+//		fmt.Printf("function: %s, args: %v", it.FnType(), it.Args())
+//	}
+type StackIterator interface {
+	// Next moves the iterator to the next function in the stack. Returns false
+	// if it reached the bottom of the stack.
+	Next() bool
+	// FnType returns the function type of the current function.
+	FnType() api.FunctionDefinition
+	// Args returns the api.ValueType encoded arguments of the current function.
```

Comment:
one thing to note: Wasm's args are _mutable_ -- the arguments are part of locals in Wasm functions and can be modified by `local.set` instruction.

Reply:
In the current implementation of the compiler `Args()` is a slice of the actual stack. Do you think it would be worth making a copy or providing a read-only proxy to avoid a potential mutation?

Reply:
maybe all we need is a note to say copy/out immediately and/or don't retain the slice? I notice unsafe slice access in other libs like fasthttp...

Reply:
Actually I see what @mathetake means. I think we should rename this to Params and clarify that the values are current, not necessarily the same as what the function was invoked with (as a note).

Reply:
Let me just rephrase to make sure I understand:

In Wasm, function parameters can be modified with a `local.set` instruction, so `Args()` may return values that are different than what the function was initially called with, so we should add a comment warning about this.

Also seems like the actual name in the Wasm spec is "parameters", not "arguments".

Does that sound right? Happy to make the changes!

Reply:
Took a stab at clarifying here https://github.com/tetratelabs/wazero/pull/1363/commits/70ec64a17b4d258b6c138b6ac2c0c89822b38fe8

Reply:
that is exactly what I wanted to see! Thanks @pelletier